src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -0,0 +1,642 @@
+        for orb in self.orbitals:
+            if isinstance(orb, APWOrbital) and orb.type.lower() in type_count.keys():
+                type_count[orb.type] += 1
+            elif isinstance(orb, APWLocalOrbital):
+                type_count['lo'] += 1
+            else:
+                type_count['other'] += 1  # other de facto operates as a catch-all
+        return type_count
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._is_normalized = False
+
+    @check_normalized
+    def normalize(self, archive: EntryArchive, logger: BoundLogger) -> None:
+        super(BasisSet).normalize(archive, logger)
+        self.n_orbitals = len(self.orbitals)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._is_normalized = False
+
+
+class MuffinTinRegion(BasisSet, Mesh):
+    """
+    Muffin-tin region around atoms, containing the augmented part of the APW basis set.
+    The latter is structured by l-channel. Each channel contains a base (S)(L)APW definition,
+    which may be extended via local orbitals.
+    """
+
+    # there are 2 main ways of structuring the APW basis set
+    # either as APW and lo in the MT region
+    # or by l-channel in the MT region
+
+    radius = Quantity(
+        type=np.float64,
+        shape=[],
+        unit='meter',
+        description="""
+        The radius descriptor of the `MuffinTin` is spherical shape.
+        """,
+    )
+
+    l_max = Quantity(
+        type=np.int32,
+        description="""
+        Maximum angular momentum quantum number that is sampled.
+        Starts at 0.
+        """,
+    )
+
+    l_channels = SubSection(sub_section=APWLChannel.m_def, repeats=True)
+
+    def _determine_apw(self) -> dict[str, int]:
+        """
+        Aggregate the APW component count in the muffin-tin region.
+        Invokes `normalize` on `l_channels`.
+        """
+        for l_channel in self.l_channels:
+            l_channel.normalize(None, None)
+
+        type_count: dict[str, int]
+        if len(self.l_channels) > 0:
+            l_channel = self.l_channels[0]
+            # dynamically determine `type_count` structure
+            type_count = l_channel._determine_apw(l_channel.orbitals)
+            for channel in self.l_channels[1:]:
+                type_count.update(channel._determine_apw())
+        return type_count
+
+    @check_normalized
+    def normalize(self, archive, logger):
+        super().normalize(archive, logger)
+
+
+class BasisSetContainer(NumericalSettings):
+    """
+    A section defining the full basis set used for representing the electronic structure
+    during the diagonalization of a Hamiltonian (component), as defined in `ModelMethod`.
+    This section may contain multiple basis set specifications under the basis_set_components,
+    each with their own parameters.
+    """
+
+    native_tier = Quantity(
```

Comment:
why not `code_specific_precision`?

Reply:
As a name? I'm fine changing it, but this has been the term used in all the communication material.

Reply:
It is ok, if you don't want to change it is fine. When first seeing this quantity I had to read the description to read what was referring to, but if you want you can leave it as `native_tier`.

In the front-end there is a mix: `code_specific_tier` :)

Reply:
> In the front-end there is a mix: code_specific_tier :)

Honestly, who decides in this stuff? No other area but C1 uses this, so why does this get changed without any heads-up?